ui/test/nuxeo-layout.test.js

```diff
@@ -27,57 +35,123 @@ const base = url.substring(0, url.lastIndexOf('/'));
+      const layout = await fixture(html`
+        <nuxeo-layout href="${base}/layouts/dummy-layout.html" model='{"text": "dummy"}'></nuxeo-layout>
+      `);
+      await flush();
+      await layoutLoad(layout);
+
+      expect(isElementVisible(layout.$.error)).to.be.false;
+      expect(layout.element.tagName).to.equal('DUMMY-LAYOUT');
+      expect(layout.element.shadowRoot.querySelector('span')).to.exist;
+      expect(layout.element.shadowRoot.querySelector('span').innerText).to.equal('dummy');
+      expect(layout.element.shadowRoot.querySelector('nuxeo-input')).to.exist;
+    });
+
+    test('Should stamp the layout when model is empty', async () => {
+      const layout = await fixture(html`
+        <nuxeo-layout href="${base}/layouts/dummy-layout.html" model=""></nuxeo-layout>
+      `);
+      await flush();
+      await layoutLoad(layout);
+
+      expect(layout.element.shadowRoot.querySelector('span')).to.exist;
+      expect(layout.element.shadowRoot.querySelector('nuxeo-input')).to.exist;
+    });
   });
 
-  test('layout invalid field invalid', async () => {
-    const layout = await fixture(html`
-      <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "invalid"}'></nuxeo-layout>
-    `);
-    const validity = await layout.validate();
-    expect(validity).to.equal(false);
+  suite('Validation', () => {
+    test('Should validate the layout when fields are valid', async () => {
+      const layout = await fixture(html`
+        <nuxeo-layout href="${base}/layouts/dummy-layout.html" model='{"text": "valid", "data": "foo"}'></nuxeo-layout>
+      `);
+      await flush();
+      await layoutLoad(layout);
+
+      const nuxeoInput = layout.element.shadowRoot.querySelector('nuxeo-input');
+      sinon.spy(nuxeoInput, 'validate');
+      sinon.spy(layout.element, 'validate');
+      sinon.spy(layout, 'validate');
+      const validity = await layout.validate();
+      expect(validity).to.be.true;
+      expect(nuxeoInput.invalid).to.be.false;
+      expect(nuxeoInput.validate.calledOnce).to.be.true;
+      expect(layout.element.validate.calledOnce).to.be.true;
+      expect(layout.validate.calledOnce).to.be.true;
+    });
+
+    test('Should invalidate the layout when required input is empty', async () => {
+      const layout = await fixture(html`
+        <nuxeo-layout href="${base}/layouts/dummy-layout.html" model=""></nuxeo-layout>
+      `);
+      await flush();
+      await layoutLoad(layout);
+
+      const nuxeoInput = layout.element.shadowRoot.querySelector('nuxeo-input');
+      sinon.spy(nuxeoInput, 'validate');
+      sinon.spy(layout.element, 'validate');
+      sinon.spy(layout, 'validate');
+      const validity = await layout.validate();
+      expect(validity).to.be.false;
+      expect(nuxeoInput.invalid).to.be.true;
+      expect(nuxeoInput.validate.calledOnce).to.be.true;
+      expect(layout.element.validate.calledOnce).to.be.false;
```

Comment:
Looking at test's description, I'm not sure if this expectation is clear.
It won't call this method because it failed on a "local" validation?

Reply:
Yes, as soon as there is one invalid field, it won't call this method. I will add a comment to make it more clear.

Reply:
Please keep in mind that not all people will look at test's comments, but sure they will look to its description. So, I would suggest to try to improve the last.